xptifw/unit_test/CMakeLists.txt

```diff
@@ -53,16 +54,26 @@ if (TARGET LLVMSupport)
   set(SUPPORT_LIB LLVMSupport)
 endif()
 
-target_link_libraries(XPTIFWUnitTests PRIVATE
-  llvm_gtest
-  llvm_gtest_main
-  xptifw
-  ${SUPPORT_LIB}
-)
+if (DEFINED LLVM_EXTERNAL_XPTIFW_SOURCE_DIR)
+  target_link_libraries(XPTIFWUnitTests PRIVATE
+    llvm_gtest
+    llvm_gtest_main
+    xptifw
+    ${SUPPORT_LIB}
```

Comment:
How is it used for the build inside llvm and at the same time isn't required in the standalone build? I don't see where the difference in *usage* comes from...

Reply:
> How is it used for the build inside llvm

That part is still answered, isn't it?

Reply:
Are you saying that the testing is different for in-tree/standalone configurations?

Reply:
In principle, they use the unit tests directly and with how it was structured before this change, they couldn't build it on their end. They should be able to do it now.

Reply:
Why can't we build it the same way when configuring in-tree?

Reply:
Maybe I am not getting the point you are trying to make because I am not a CMake expert. So, if you have suggestions on this, please do so and will consider making the change.

Reply:
I'm not a CMake expert either and my question has little to do with CMake. What I don't understand is why you have to link extra library in the in-tree configuration. What would happen if you'd just use the one without `SUPPORT_LIB` for both scenarios?

Reply:
I do not know what SUPPORT_LIB brings to the equation, and we may have to look at the history of when it was added and why it was added. It may be the case of needing libstdc++ if the in-tree build was configured as such.